analysis-pipeline/predictability/fit.conseq

```diff
@@ -0,0 +1,134 @@
+        confounder={"type": "feature", "label": "confounder"},
+        driver_events={"type": "feature", "label": "driver_events"},
+        model_config=fileref("model-config.yaml")
+    outputs:
+        {
+        "type": "input-model-file",
+        "filename": {"$filename": "model-map.json"}
+        },
+
+    run "python" with """
+    import json
+    import yaml
+    import subprocess
+    import os
+
+    model_config_file = "{{ inputs.model_config.filename }}"
+    with open(model_config_file, 'r') as file:
+        config = yaml.safe_load(file)
+    
+    # Process each model in the YAML
+    for model_name, model_config in config.items():
+        output_json = {
+            "model_name": model_name,
+            "screen_name": model_config["Screen"],
+            "data": {}
+        }
+        
+        target = model_config["Target"][0]
+
+        # Choose the correct target based on screen name
+        if model_config["Screen"] == "CRISPR":
+            target_id = "{{ inputs.crispr_gene_effect.source_dataset_id }}"
+        elif model_config["Screen"] == "RNAi":
+            target_id = "{{ inputs.rnai.source_dataset_id }}"
+
+        output_json["data"][target] = {
+            "taiga_id": target_id,
+            "table_type": "target_matrix",
+            "relation": model_config["Relation"]
+        }
+        
+        for feature in model_config["Features"]:
+            if feature.lower() == "lineage":
+                feature_input = {{ inputs.lineage }}
+            elif feature.lower() == "confounder":
+                feature_input = {{ inputs.confounder }}
+            elif feature.lower() == "driverevents":
+                feature_input = {{ inputs.driver_events }}
+            
+            output_json["data"][feature] = {
+                "taiga_id": feature_input["source_dataset_id"],
+                "table_type": feature_input["type"],
+                "dim_type": feature_input["label"].lower(),
+                "required": feature in model_config["Required"],
+                "exempt": False
+            }
+        
+        output_file = f"{model_name.lower()}_config.json"
+        output_file_path = os.path.abspath(output_file)
+
+        with open(output_file, 'w') as f:
+            json.dump(output_json, f, indent=2)
+        print(f"Created {output_file}")
+
+        
+        docker_command = [
+            "docker", "run",
+            "-w", "/daintree",
+            "--pull=always",
+            "-v", f"{output_file_path}:/daintree/{output_file}",
+            "us.gcr.io/broad-achilles/daintree-sparkles:v4",
+            "/install/depmap-py/bin/python3.9", "-u", "run_fit_models.py",
+            "collect-and-fit-generate-config",
+            "--input-files", output_file,
+            "--sparkles-path", "/install/sparkles/bin/sparkles",
+            "--sparkles-config", "/daintree/sparkles-config",
+            "--save-dir", "/daintree/output_data/nayeem-test",
```

Comment:
I don't see /daintree/output_data being mounted from outside of the docker image. That suggests the output will get thrown away when the container is deleted.

Instead of mounting individual files, I often find it easier to mount the current working directory to some location in the container and use that for input and output.